python/ray/data/preprocessor.py

```diff
@@ -187,36 +188,63 @@ def _fit(self, dataset: Dataset) -> "Preprocessor":
     def _determine_transform_to_use(self, data_format: str) -> str:
         """Determine which transform to use based on data format and implementation.
 
-        * If only _transform_arrow is implemented, will convert the data to arrow.
-        * If only _transform_pandas is implemented, will convert the data to pandas.
-        If both are implemented, will pick the method corresponding to the format
-        for best performance.
-        * Implementation is defined as overriding the method in a sub-class.
+        * If batch_format is given as:
+            * numpy and _transform_numpy is implemented will transform as numpy.
+            * pandas and _transform_pandas is implemented will transform as pandas.
+            * If both are implemented, we respect the user's choice of batch_format.
+        * Otherwise, we will infer and pick the best transform to use:
+            * pandas data format prioritized transform as pandas if available.
+            * arrow data format prioritized transform as numpy if available.
```

Comment:
should this say "numpy data format"?

Reply:
we have a `dataset.from_numpy` API but internally there's no data block format of numpy -- they're converted into either pandas or arrow

Reply:
actually for single column dataset we *could* return `numpy` format. I've updated the docstring accordingly.